### ISSUE TITLE
Can not use collect subqueries in THEN clause

### ISSUE THREAD

**Issue by reckter** [neo4j/neo4j#13118]
To help us understand your issue, please specify important details, primarily:

- Neo4j version: 5.5.0
- Operating system: Mac
- API/Driver: Web interface

Using the following query:
```cypher
match (a:Something)
with a,
    CASE
        WHEN true
            THEN COLLECT {
                 MATCH (a) -> (b:SomethingElse)
                 RETURN b.foo
            }
        ELSE null
    END AS test
RETURN test
```
I get the syntax error:
```
Invalid input '(': expected "," or "}" (line 6, column 23 (offset: 104))
"                MATCH (a) -> (b:SomethingElse)"
                       ^                  
```

Looking at the documentation for Case expressions here (https://neo4j.com/docs/cypher-manual/current/syntax/expressions/#query-syntax-case), it should be possible to use a `COLLECT subquery` there.

**Comment by AzuObs** [comment]
Hello Hannes,

Thank you for reporting the issue. 

You are right, you can use the collect subquery in this position. The problem is with the pattern within your collect subquery. `(a) -> (b:SomethingElse)` is not a valid pattern, but `(a) --> (b:SomethingElse)` would be.

Whole query (fixed)
```
match (a:Something)
with a,
    CASE
        WHEN true
            THEN COLLECT {
                 MATCH (a)-->(b:SomethingElse)
                 RETURN b.foo
            }
        ELSE null
    END AS test
RETURN test
```

Hope this solves your issue.

**Comment by reckter** [comment]
@AzuObs 
`-->` vs `->` doesn't seem to solve the issue: 
```
Invalid input '(': expected "," or "}" (line 6, column 24 (offset: 105))
"                 MATCH (a)-->(b:SomethingElse)"
                        ^
```

(tried changing it myself and copied your suggestion, both hat the same error)

Seems that also was an issue testing the `->` vs `-->` in it's own query, but doesn't seem to be the only problem in this case :)

**Comment by AzuObs** [comment]
Apologies, I was originally running my query in Neo4j 5.6.0. I tried my example in 5.5.0 and I also have this problem. Below I explain why the problem is occurring, and how to fix it.

[COLLECT () Aggregation Functions](https://neo4j.com/docs/cypher-manual/current/functions/aggregating/#functions-collect) have existed for some time and are available in 5.5.0.

[COLLECT {} Subqueries](https://neo4j.com/docs/cypher-manual/current/syntax/expressions/#collect-subqueries) have been introduced in 5.6.0.

These constructs have different syntaxes, and the 5.5.0 Cypher compiler is unhappy that you are trying to use `{` after the COLLECT keyword because it doesn't recognise it.

Would it be possible for you to upgrade your database to 5.6.0? I think that we are releasing 5.7.0 later today, which should work. Any future dbms version should also work.

Warm thanks,

**Comment by Lojjs** [comment]
@reckter @AzuObs I will close this Gihub issue now, as there does not seem to be anything for us in Cypher team to fix. Feel free to re-open if there is still a problem.

Best regards Louise
